File: SportsStoreVb/SportsStoreVb.Domain/Concrete/EFProductRepository.vb
Imports SportsStoreVb.Domain.SportsStoreVb.Domain.Entities

Namespace SportsStoreVb.Domain.Concrete
    Public Class EFProductRepository
        Implements IProductRepository

        Private context As New EFDbContext()

        Public ReadOnly Property Products As IEnumerable(Of Product) Implements IProductRepository.Products
            Get
                Return context.Products
            End Get
        End Property
    End Class
End Namespace

File: SportsStoreCs/SportsStoreCs.Domain/Concrete/EFProductRepository.cs
namespace SportsStoreCs.Domain.Concrete
{
    using System.Collections.Generic;
    using Abstract;
    using Entities;

    public class EFProductRepository : IProductRepository
    {
        EFDbContext context = new EFDbContext();

        public IEnumerable<Product> Products
        {
            get { return context.Products;}
        }
    }
}